content/hardware/06.nicla/boards/nicla-voice/tutorials/ei-intruder-detector/content.md

```diff
@@ -0,0 +1,474 @@
+
+### Nicla Voice Code
+
+Now, we will go through some important code sections to make this application fully operative. We will begin with the required libraries:
+
+- Including `NDP.h` will enable and run the Neural Decision Processor (NDP120), it's included in the BSP of the Nicla Voice.
+- Including `ArduinoBLE.h` will enable the BLE communication, install it by searching for it on the Library Manager.
+
+The BLE services and characteristics are standardized for the specific use of this application's features. The service is defined as "Immediate alert" which makes it ideal for the use that we will give of notifying on a door opening or forcing event, in addition to this, we defined two characteristics, one for the "Alert Level" that will advertise between both events and a "Battery Level" one to advertise the Nicla Voice battery level to the host. Notice that they have specific and standardized UUIDs.
+
+```arduino
+#include "NDP.h"
+#include <ArduinoBLE.h>
+
+// Alert Service
+BLEService alertService("1802");  // Immediate alert
+
+// BLE Alert Characteristic
+BLEUnsignedCharCharacteristic alertLevel("2A06",                // standard 16-bit characteristic UUID
+                                         BLERead | BLENotify);  // remote clients will be able to get notifications if this characteristic changes
+
+// Bluetooth® Low Energy Battery Level Characteristic
+BLEUnsignedCharCharacteristic batteryLevelChar("2A19",                // standard 16-bit characteristic UUID
+                                               BLERead | BLENotify);  // remote clients will be able to get notifications if this characteristic changes
+```
+The Nicla Voice's integrated NDP needs some files stored in the external flash of the board which are:
+- mcu_fw_120_v91.synpkg
+- dsp_firmware_v91.synpkg
+- ei_model.synpkg
+
+> The files must be named exactly as showed below.
+To store them in the Nicla Voice memory you must use the [Syntiant Uploader](assets/Syntiant_Uploader.zip), these files won't be overwritten after any sketch update.
+
+
+```arduino
+  // Neural Decision Processor firmware and ML model files loading
+  NDP.begin("mcu_fw_120_v91.synpkg");
+  NDP.load("dsp_firmware_v91.synpkg");
+  NDP.load("ei_model.synpkg");
+  Serial.println("packages loaded");
+  NDP.getInfo();
+  Serial.println("Configure mic");
+  NDP.turnOnMicrophone();
+```
+The main responsibility of the Nicla Voice code is to listen to and identify the trained sounds, the below code section is in charge of comparing the inferred category and taking a certain action between them. In the case the Nicla detects the door opening, the label parameter will turn to "NN0:opened", this will trigger a BLE alert sending command followed by a listening pause to avoid duplicated alerts and a green LED blinking to visually indicate the event. The same for the "NN0:forcing" label, with the difference that in this case the LED flashes red and the alert message changes.
```

Comment:
```suggestion
The main responsibility of the Nicla Voice code is to listen to and identify the trained sounds. The below code section is in charge of comparing the inferred category and taking a certain action between them. In the case the Nicla detects the door opening, the label parameter will turn to "NN0:opened", this will trigger a BLE alert sending command followed by a listening pause to avoid duplicated alerts and a green LED blinking to visually indicate the event. The same for the "NN0:forcing" label, with the difference that in this case the LED flashes red and the alert message changes.
```

Punctuation suggestion